snaps_k8s/provision/ansible_configuration.py

```diff
@@ -902,3 +929,189 @@ def delete_weave_interface(k8s_conf):
+    if not(ret_val_lb_config and ret_val_ha_config):
+        logger.error('Alert !! Failed in HA Configuration')
+        exit(1)
+                        
+def __launch_ha_loadbalancer_conf(hostname_map=None,
+                                  host_node_type_map=None,
+                                  loadbalancer_dict=None,
+                                  project_name=None,pb_vars=None):
+    """
+    function used to call launch_load_balancer
+    :param hostname_map:
+    :param host_node_type_map:
+    :param loadbalancer_dict:
+    :param project_name:
+    :return:
+    """
+    logger.info("Argument List:\n hostname_map: %s\n host_node_type_map: "
+                "%s\n loadbalancer_dict: %s\n project_name: %s\n pb_vars: %s",
+                hostname_map, host_node_type_map, loadbalancer_dict,
+                project_name,pb_vars)
+
+    ret_val = False
+    logger.info("launch_ha_loadbalancer_conf function")
+    host_node_type_map_local = {}
+    lb_port = str(loadbalancer_dict.get("port"))
+    master_ip_list = []
+    for key in host_node_type_map:
+        if host_node_type_map[key] == "master":
+            master_ip_list.append(hostname_map[key])
+    master_ip_list = str(master_ip_list)
+    logger.info(host_node_type_map_local)
+    logger.info('INSTALL / CONFIGURE EXTERNAL LOAD BALANCER on -- ip %s',
+                            host_node_type_map_local.get('localhost'))
+    try:
+        ret_val = __launch_load_balancer(
+            consts.K8_HA_EXT_LB,
+            loadbalancer_dict.get(consts.IP_KEY),
+            loadbalancer_dict.get(consts.HOSTNAME_KEY),
+            master_ip_list, lb_port,
+            loadbalancer_ip=loadbalancer_dict.get(consts.IP_KEY),
+            project_name=project_name,
+            base_pb_vars_ha=pb_vars)
+    except Exception as exception:
+        logger.error(exception.message)
+        logger.info("exception while calling "
+                    "external_load_balancer_install ")
+        ret_val = False
+    return ret_val
+
+
+def __launch_kubespray_ha_configure(hostname_map=None,
+                                    host_node_type_map=None,
+                                    loadbalancer_dict=None,
+                                    project_name=None,pb_vars=None):
+    """
+    fucntion used to call kubespray_ha_configure
+    :param hostname_map:
+    :param host_node_type_map:
+    :param loadbalancer_dict:
+    :param project_name:
+    :return:
+    """
+    logger.info("Argument List:\n hostname_map: %s\n host_node_type_map: "
+                "%s\n loadbalancer_dict: %s\n project_name: %s",
+                hostname_map, host_node_type_map, loadbalancer_dict,
+                project_name)
+
+    ret_val = False
+    logger.info("launch_ha_kubespray_configure function")
+    host_node_type_map_local = {'localhost': '127.0.0.1'}
+    try:
+        ret_val = __kubespray_ha_configure(
```

Comment:
Do not store return value. Exception handling will deal with this properly

Reply:
Incorporated.